screens/ActivitiesScreen.tsx

```diff
@@ -1,15 +1,174 @@
-import React from "react"
+import EventsList from "@components/EventsList"
+import EventsMap, { MapRefMethods } from "@components/EventsMap"
+import { Ionicon } from "@components/common/Icons"
+import EventTabBar from "@components/tabBarComponents/EventTabBar"
+import { useTrackUserLocation } from "@hooks/UserLocation"
+import { CurrentUserEvent, EventMocks } from "@lib/events"
+import {
+  UserLocationTrackingUpdate,
+  requestLocationPermissions
+} from "@lib/location/UserLocation"
+import React, { useEffect, useRef } from "react"
+import {
+  ImageBackground,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View
+} from "react-native"
+import { Icon } from "react-native-elements"
 import EventDetails from "./EventDetails/EventDetails"
 import { EventMocks } from "@lib/events"
 
-const MARKER_SIZE = 80
+const MARKER_SIZE = 60
 
 const ActivitiesScreen = () => {
+  const appRef = useRef<MapRefMethods | null>(null)
+  let doesWork = false
+  const givenUserLocation = useTrackUserLocation()
+
+  const recenterThing = () => {
+    if (givenUserLocation?.status === "success") {
+      appRef.current?.recenterToLocation({
+        latitude: givenUserLocation?.location.coordinates.latitude,
+        longitude: givenUserLocation?.location.coordinates.longitude
+      })
+    }
+  }
+  const events: CurrentUserEvent[] = [
+    EventMocks.Multiday,
+    EventMocks.NoPlacemarkInfo,
+    EventMocks.PickupBasketball
+  ]
+
+  useEffect(() => {
+    ;(async () => {
+      const status = await requestLocationPermissions()
+      if (status === false) {
+      } else {
+        doesWork = true
+      }
+    })()
+  }, [])
+
+  function checkUserLocation (
+    userLocationParam: UserLocationTrackingUpdate | undefined
+  ) {
+    if (userLocationParam?.status === "success") {
+      return {
+        latitude: userLocationParam.location.coordinates.latitude,
+        longitude: userLocationParam.location.coordinates.longitude,
+        latitudeDelta: 0.5,
+        longitudeDelta: 0.5
+      }
+    } else {
+      return {
+        latitude: 30.0,
+        longitude: 30.0,
+        latitudeDelta: 0.5,
+        longitudeDelta: 0.5
+      }
+    }
+  }
+
+  console.log(givenUserLocation)
   return (
     <>
-      <EventDetails event={EventMocks.NoPlacemarkInfo}/>
-    </>
+      <EventsMap
```

Comment:
I think it's best that you just make all of this map recentering logic and what not a separate component itself. Actually, it could probably be placed inside the `EventsMap`.